pkg/service/service.go

```diff
@@ -861,6 +869,35 @@ func (s *Service) restoreBackendsLocked() error {
 			logfields.BackendID: b.ID,
 			logfields.L3n4Addr:  b.L3n4Addr.String(),
 		}).Debug("Restoring backend")
+		if _, ok := svcBackendsById[b.ID]; !ok && s.backendRefCount[b.L3n4Addr.Hash()] != 0 {
+			// If a backend by id isn't referenced by any of the service entries,
+			// it's likely to be a duplicate backend. This can happen when agent
+			// leaked backend entries in the backends map prior to restart, and created
+			// duplicate with different IDs but same L3n4Addr (hash).
+			// As none of the service entries have a reference to these backends
+			// in the services map, the duplicate backends were not available for
+			// load-balancing new traffic. While there is a slim chance that the
+			// duplicate backends could have previously established active connections,
+			// and these connections can get disrupted. However, the leaks likely
+			// happened when service entries were deleted, so those connections
+			// were also expected to be terminated.
+			// Regardless, delete the duplicates as this can affect restoration of current
+			// active backends, and may prevent new backends getting added as map
+			// size is limited, which can result in connectivity issues.
+			id := b.ID
+			DeleteBackendID(id)
+			if err := s.lbmap.DeleteBackendByID(id); err != nil {
```

Comment:
```suggestion
			if err := s.lbmap.DeleteBackendByID(id, b.L3n4Addr.IsIPv6()); err != nil {
```

Reply:
The one backport PR where I don't run `git rebase --exec "make build" origin/v1.11` :facepalm: Thanks for the pointers Aditi!